src/components/Form.js

```diff
@@ -1,34 +1,67 @@
-import { useState } from 'react'
+import { useState, useEffect, createRef } from 'react'
 import styled from 'styled-components/macro'
 import PropTypes from 'prop-types'
 import Button from './Button'
 import Input from './Input'
+import { CrossIcon } from '../lib/Icons'
 
 function Form({ userPlace, setUserPlace, status }) {
     const [userInput, setUserInput] = useState('')
+    const [validation, setValidation] = useState(true)
+
+    const textInput = createRef()
+
+    useEffect(() => setValidation(true), [userInput])
 
     function handleChange(event) {
         setUserInput(event.target.value)
     }
 
     function handleSubmitPlace(event) {
         event.preventDefault()
-        userInput === userPlace
-            ? setUserPlace(userInput + ' ')
-            : setUserPlace(userInput)
+        if (validateInput()) {
+            userInput === userPlace
+                ? setUserPlace(userInput + ' ')
+                : setUserPlace(userInput.trim())
+            setUserInput('')
+        } else {
+            setValidation(false)
+        }
+    }
+
+    function validateInput() {
+        const regexCountyCode = /[0-9]/g
+        const regexPlaceName = /[^a-z-\s]/gi
+
+        if (!regexPlaceName.test(userInput)) {
+            return userInput.length >= 2 && userInput.length <= 32
+        } else {
+            return regexCountyCode.test(userInput) && userInput.length === 5
+        }
```

Comment:
Not an issue and only code style related, but you can remove `else` here, because you `return` in `if`.

```suggestion
        } 
        return regexCountyCode.test(userInput) && userInput.length === 5
```